lib/min_heap.rb

```diff
@@ -14,18 +14,23 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: 0(log n)
+  # Space Complexity: 0(1)
```

Comment:
Since `heap_down` is recursive you have to consider the impact of the recursion on space complexity.